x/emissions/keeper/inference_synthesis/network_inference_builder.go

```diff
@@ -218,189 +361,507 @@ func (b *NetworkInferenceBuilder) SetOneOutInfererValues() *NetworkInferenceBuil
 	}
-	b.oneOutForecasterInferences = oneOutImpliedInferences
-	return b
+	return oneOutForecasterInferences, nil
+}
+
+// Arguments to calcOneInValue
+type calcOneInValueArgs struct {
+	Ctx                                  sdk.Context
+	K                                    emissionskeeper.Keeper
+	Logger                               log.Logger
+	TopicId                              uint64
+	AllInferersAreNew                    bool
+	Inferers                             []Inferer
+	InfererToInference                   map[Inferer]*emissions.Inference
+	ForecasterToForecast                 map[Forecaster]*emissions.Forecast
+	ForecasterToForecastImpliedInference map[Forecaster]*emissions.Inference
+	EpsilonTopic                         alloraMath.Dec
+	EpsilonSafeDiv                       alloraMath.Dec
+	PNorm                                alloraMath.Dec
+	CNorm                                alloraMath.Dec
+	OneInForecaster                      Forecaster
 }
 
-func (b *NetworkInferenceBuilder) calcOneInValue(oneInForecaster Worker) (alloraMath.Dec, error) {
-	b.logger.Debug(fmt.Sprintf("Calculating one-in inference for forecaster: %s", oneInForecaster))
-	palette := b.palette.Clone()
+// Calculate the one-in inference given a withheld forecaster
+func calcOneInValue(args calcOneInValueArgs) (
+	oneInInference alloraMath.Dec,
+	err error,
+) {
+	args.Logger.Debug(fmt.Sprintf("Calculating one-in inference for forecaster: %s", args.OneInForecaster))
 
 	// In each loop, remove all forecast-implied inferences except one
-	forecastImpliedInferencesWithForecaster := make(map[Worker]*emissions.Inference)
-	forecastImpliedInferencesWithForecaster[oneInForecaster] = palette.ForecastImpliedInferenceByWorker[oneInForecaster]
-	palette.ForecastImpliedInferenceByWorker = forecastImpliedInferencesWithForecaster
+	singleForecastImpliedInference := make(map[Worker]*emissions.Inference, 1)
+	singleForecastImpliedInference[args.OneInForecaster] = args.ForecasterToForecastImpliedInference[args.OneInForecaster]
 
 	// Get self regret for the forecaster
-	regret, _, err := palette.K.GetOneInForecasterNetworkRegret(palette.Ctx, palette.TopicId, oneInForecaster, oneInForecaster)
+	singleForecasterRegret := make(map[Worker]*Regret, 1)
+	regret, _, err := args.K.GetOneInForecasterNetworkRegret(args.Ctx, args.TopicId, args.OneInForecaster, args.OneInForecaster)
 	if err != nil {
-		return alloraMath.Dec{}, errorsmod.Wrapf(err, "Error getting one-in forecaster regret")
+		return alloraMath.Dec{}, errorsmod.Wrapf(err, "CalcOneInValue() error getting one-in forecaster regret")
 	}
+	singleForecasterRegret[args.OneInForecaster] = &regret.Value
 
-	palette.ForecasterRegrets[oneInForecaster] = &regret.Value
+	// get self forecast list
+	singleForecaster := []Worker{args.OneInForecaster}
 
-	remainingForecaster := []Worker{oneInForecaster}
-	err = palette.UpdateForecastersInfo(remainingForecaster)
```

Comment:
Surprising but welcome compute savings here and in `UpdateInferersInfo()` call below. good stuff